src-tauri/src/lib.rs

```diff
@@ -1,14 +1,193 @@
-// Learn more about Tauri commands at https://tauri.app/develop/calling-rust/
+use serde::{Deserialize, Serialize};
+use std::fs;
+use tokio::fs::File;
+use tokio::io::AsyncWriteExt;
+
+// Struct representing the latest versions of different release types
+#[derive(Debug, Serialize)]
+struct LatestVersions {
+    nightly: Option<String>,
+    stable: Option<String>,
+    alpha: Option<String>,
+    beta: Option<String>,
+    rc: Option<String>,
+    error: bool,
+}
+
+impl LatestVersions {
+    // Constructor for LatestVersions, initializing all versions to None and setting the error flag
+    fn new(error: bool) -> Self {
+        Self {
+            nightly: None,
+            stable: None,
+            alpha: None,
+            beta: None,
+            rc: None,
+            error,
+        }
+    }
+}
+
+// Struct representing the tag response from the GitHub API
+#[derive(Deserialize)]
+struct TagResponse {
+    pub r#ref: String,
+}
+
 #[tauri::command]
-fn greet(name: &str) -> String {
-    format!("Hello, {}! You've been greeted from Rust!", name)
+// Asynchronous function to fetch the latest versions from the GitHub API
```

Comment:
Annotations should not declared under macros, I will setup a `clippy` workflows later to check your codes.